R/as_wide_table.R

```diff
@@ -15,14 +15,14 @@
 #' @importFrom utils methods
 
 as_wide_table <- function(austraits){
-  # Switch for different versions
-  version <- what_version(austraits)
+  # Check compatability
+  status <- check_compatibility(austraits)
   
-  switch (version,
-          "5-series" = as_wide_table3(austraits),
-          "4-series" = as_wide_table2(austraits),
-          "3-series-earlier" = as_wide_table1(austraits)
-          )
+  # If compatible
+  if(!status){
+    function_not_supported(austraits)
+  }
+  as_wide_table3(austraits)
```

Comment:
should we name these based on the versions they support? I.e., even though it's function variant number 3, we could name it `as_wide_table_v5`. Not sure, just floating the idea.

Reply:
Will further tear these sub functions down so that we are maintaining 1 function